### PR TITLE
FEATURE: add a `path add` to the standard library

### PR DESCRIPTION
**Pull request by amtoine**
# Description
this PR adds the `path add` command to `crates/nu-utils/standard_library/std.nu`
- this comes from frequent questions over on the discord server, about how to add directories to the `PATH`
- this is greatly inspired from the [original `path-add`](https://discord.com/channels/601130461678272522/615253963645911060/1081206660816699402) from @melMass 
- allows to prepend and append a variable number of directories to the `PATH`
- i've added a description with an example
- i've added tests in `crates/nu-utils/standard_library/tests.nu` that hopefully covers all the features

# User-Facing Changes
`path add` can now be used from `std.nu`

# Tests + Formatting
the tests pass with
```bash
nu crates/nu-utils/standard_library/tests.nu
```

# After Submitting
```bash
$nothing
```

### CODE REVIEW

**Comment by kubouch** [comment]
I think this is a good idea, but I'm worried the name collides with the existing `path` commands. We used to have a `pathvar` command long time ago, maybe that could be used? Although it's quite long, there might be a better name.

**Comment by amtoine** [comment]
> I think this is a good idea
>
:yum: 

> but I'm worried the name collides with the existing `path` commands.
>
there is no `path add` for now, right, and this would be called `std path add` in the end :thinking: 
but yeah there is a chance of collision when doing `use .../std.nu "path add"`...

if it's just the `path` overlap, i do not think this is a bad thing!
i often "add" subcommands to already existing commands, e.g. `path add extension` to add a file extension to a path, `cargo list` to list installed crates as structured data, etc, etc...

> We used to have a `pathvar` command long time ago, maybe that could be used? Although it's quite long, there might be a better name.
>
but yeah, i agree, there might be a better name :confused: 

**TL;DR**: as there is no builtin `path add` for now, i think this is ok to leave it as is.
however, we should wait for other feedbacks and maybe new name ideas :wink:
